test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -283,3 +292,84 @@
+                          (init [this context] (lc-fn context :init-bonus-service))
+                          (start [this context] (lc-fn context :start-bonus-service))
+                          (stop [this context] (lc-fn context :stop-bonus-service))
+                          (bonus-service-fn [this] (lc-fn nil :bonus-service-fn)))]
+      (bootstrap/with-puppetserver-running-with-services
+       app
+       (conj (tk-bootstrap/parse-bootstrap-config! bootstrap/dev-bootstrap-file) bonus-service)
+       {:jruby-puppet {:max-active-instances 1}}
+       (tk-internal/register-sighup-handler [app])
+       (beckon/raise! "HUP")
+       (let [start (System/currentTimeMillis)]
+         (while (and (not= (count @call-seq) 5)
+                     (< (- (System/currentTimeMillis) start) 90000))
+           (Thread/yield)))
+       ; 5 elements in the call-seq corresponds to init, start, stop, init, start for the BonusService
+       (is (= (count @call-seq) 5))
+       (let [get-results (http-client/get "https://localhost:8140/puppet/v3/environments"
+                                          bootstrap/request-options)]
+         (is (= 200 (:status get-results))))))))
+
+(deftest ^:integration test-503-when-app-shuts-down
+  (testing "During a shutdown the agent requests result in a 503 response"
+    (let [services [jruby/jruby-puppet-pooled-service profiler/puppet-profiler-service
+                    handler-service/request-handler-service ps-config/puppet-server-config-service
+                    jetty9/jetty9-service]
+          config (-> (jruby-testutils/jruby-puppet-tk-config
+                      (jruby-testutils/jruby-puppet-config {:max-active-instances 2}))
+                     (assoc-in [:webserver :port] 8081))
+          app (tk/boot-services-with-config services config)
+          cert (ssl-utils/pem->cert
+                (str test-resources-dir "/localhost-cert.pem"))
+          jruby-service (tk-app/get-service app :JRubyPuppetService)
+          jruby-instance (jruby-protocol/borrow-instance jruby-service :i-want-this-instance)
+          handler-service (tk-app/get-service app :RequestHandlerService)
+          request { :uri "/puppet/v3/environments", :params {}, :headers {},
+                   :request-method :GET, :body "", :ssl-client-cert cert, :content-type ""}
+          ping-environment #(->> request (handler-core/wrap-params-for-jruby) (handler/handle-request handler-service))
+          stop-complete? (future (tk-app/stop app))]
+        (let [start (System/currentTimeMillis)]
+          (while (and
+                  (< (- (System/currentTimeMillis) start) 10000)
+                  (not= 503 (:status (ping-environment))))
+            (Thread/yield))
+          (is (= 503 (:status (ping-environment))))
+          (jruby-protocol/return-instance jruby-service jruby-instance :i-want-this-instance)
+          @stop-complete?
+          (is (= 503 (:status (ping-environment))))))))
```

Comment:
How does this work?  The app should be shutdown at this point in the test, right?